sensor/common/sensor/central_communication_impl.go

```diff
@@ -242,7 +253,36 @@ func (s *centralCommunicationImpl) initialSync(stream central.SensorService_Comm
 		return err
 	}
 
-	return s.initialPolicySync(stream, detector)
+	if err := s.initialPolicySync(stream, detector); err != nil {
+		return err
+	}
+
+	return s.initialDeduperSync(stream)
+}
+
+func (s *centralCommunicationImpl) initialDeduperSync(stream central.SensorService_CommunicateClient) error {
+	// If client reconciliation is disabled or cental does not support it, don't expect a deduper sync message to arrive
+	if !s.clientReconcile || !centralcaps.Has(centralsensor.SensorReconciliationOnReconnect) {
+		log.Info("Skipping client reconciliation. Sensor will not wait for deduper state")
+		return nil
+	}
+	log.Info("Waiting for deduper state from Central")
+
+	msg, err := stream.Recv()
+	if err != nil {
+		if e, ok := status.FromError(err); ok {
+			if e.Code() == codes.ResourceExhausted {
+				return errors.Wrap(errCantReconcile, e.String())
+			}
+		}
+		return errors.Wrap(err, "receiving initial deduper sync")
+	}
+	if msg.GetDeduperState() == nil {
```

Comment:
Would it make sense to not reconcile (in sensor) in the reconnect if we go down this path?